lib/file-source/src/file_server.rs

```diff
@@ -148,11 +148,11 @@ where
                 let emitter = emitter.clone();
                 let checkpointer = Arc::clone(&checkpointer);
                 tokio::task::spawn_blocking(move || {
-                    checkpointer
-                        .write_checkpoints()
-                        .map_err(|error| emitter.emit_file_checkpoint_write_failed(error))
-                        .map(|count| emitter.emit_file_checkpointed(count))
-                        .ok()
+                    let start = time::Instant::now();
+                    match checkpointer.write_checkpoints() {
+                        Ok(count) => emitter.emit_file_checkpointed(count, start.elapsed()),
```

Comment:
:+1: to removing this map usage. I always find it a bit awkward when map is used to "consume" values.